pkg/audit/audit.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"io"
+
+	"github.com/vivint/infectious"
+	"google.golang.org/grpc"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/piecestore/rpc/client"
+	"storj.io/storj/pkg/provider"
+)
+
+func setupPSClient(ctx context.Context) (psClient client.PSClient, err error) {
+	// TODO(nat): get this info from the satellite somehow
+	ca, err := provider.NewCA(ctx, 12, 4)
+	if err != nil {
+		return nil, err
+	}
+	identity, err := ca.NewIdentity()
+	if err != nil {
+		return nil, err
+	}
+	identOpt, err := identity.DialOption()
+	if err != nil {
+		return nil, err
+	}
+
+	var conn *grpc.ClientConn
+	conn, err = grpc.Dial(":7777", identOpt)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	psClient, err = client.NewPSClient(conn, 1024*32, identity.Key.(*ecdsa.PrivateKey))
+	if err != nil {
+		return nil, err
+	}
+	return psClient, nil
+}
+
+func getShare(ctx context.Context, stripeIndex, required, shareSize int,
+	id client.PieceID, segSize int64) (share infectious.Share, err error) {
+	psClient, err := setupPSClient(ctx)
+	if err != nil {
+		return infectious.Share{}, err
+	}
+
+	rr, err := psClient.Get(ctx, id, segSize, &pb.PayerBandwidthAllocation{})
+	if err != nil {
+		return infectious.Share{}, err
+	}
+
+	offset := shareSize*stripeIndex - 1
+
+	rc, err := rr.Range(ctx, int64(offset), int64(shareSize))
+	if err != nil {
+		return infectious.Share{}, err
+	}
+
+	buf := make([]byte, shareSize)
+	_, err = io.ReadFull(rc, buf)
+	if err != nil {
+		return infectious.Share{}, err
+	}
+
+	share = infectious.Share{
+		Number: stripeIndex,
```

Comment:
This will not do what you are intending. `stripeIndex` will be the same for each separate piece you are getting an eshare from. Let's say you're downloading a stripe with an index 2 and that stripe was erasure encoded into shares using a 20/40 scheme. Each of the eshares for that stripe (40) would be uploaded to a different storage node, but all those eshares would have the same `stripeIndex`. Meaning each of your infectious shares for the audit of this stripe will have the same `Number` value. 

I think you should have another `int` arg in this function called `shareNumber` or something and just put it here instead of `stripeIndex`